src/stores/useGovStore.ts

```diff
@@ -29,20 +37,89 @@ export const useGovStore = defineStore('govStore', {
     initial() {
       this.$reset();
       this.fetchParams();
-      this.fetchProposals('2');
+      // this.fetchProposals to be run on page load (onMounted)
     },
+
     async fetchProposals(status: string, pagination?: PageRequest) {
       //if (!this.loading[status]) {
       this.loading[status] = LoadingStatus.Loading;
-      const proposals = reactive(
-        await this.blockchain.rpc?.getGovProposals(status, pagination)
-      );
+
+      // Check cache first
+      const cached = GovProposalCache.get(this.blockchain.chainName, status);
+
+      let proposals;
+      let lastKnownProposalId = 0;
```

Comment:
Consider storing this as a part of the cache, so you don't have to run `Math.max(...cached.proposals.map(p => parseInt(p.proposal_id)));` each time.